tests/integration/level_depth/test_search_different_depths.py

```diff
@@ -24,31 +18,41 @@ def test_index_depth_0_search_depth_1():
     def validate_granularity_1(resp):
         assert len(resp.docs) == 3
         for doc in resp.docs:
-            assert doc.granularity == 1
-            assert len(doc.matches) == 1
+            assert doc.granularity == 0
+            assert len(doc.matches) == 3
             assert doc.matches[0].id == doc.id  # done on purpose
             assert doc.matches[0].granularity == 0
 
-        assert resp.docs[0].text == 'I am chunk 1 of doc 1,'
-        assert resp.docs[0].matches[0].text == 'I am chunk 0 of doc 1, I am chunk 1 of doc 1, I am chunk 2 of doc 1'
+        assert resp.docs[0].text == ' I am chunk 1 of doc 1,'
+        assert (
+            resp.docs[0].matches[0].text
+            == 'I am chunk 0 of doc 1, I am chunk 1 of doc 1, I am chunk 2 of doc 1'
+        )
 
         assert resp.docs[1].text == 'I am chunk 0 of doc 2,'
-        assert resp.docs[1].matches[0].text == 'I am chunk 0 of doc 2, I am chunk 1 of doc 2'
+        assert (
+            resp.docs[1].matches[0].text
+            == 'I am chunk 0 of doc 2, I am chunk 1 of doc 2'
+        )
 
-        assert resp.docs[2].text == 'I am chunk 3 of doc 3'
-        assert resp.docs[2].matches[
-                   0].text == 'I am chunk 0 of doc 3, I am chunk 1 of doc 3, I am chunk 2 of doc 3, I am chunk 3 of doc 3'
+        assert resp.docs[2].text == ' I am chunk 3 of doc 3'
+        assert (
+            resp.docs[2].matches[0].text
+            == 'I am chunk 0 of doc 3, I am chunk 1 of doc 3, I am chunk 2 of doc 3, I am chunk 3 of doc 3'
+        )
 
     search_data = [
-        'I am chunk 1 of doc 1,',
+        ' I am chunk 1 of doc 1,',
         'I am chunk 0 of doc 2,',
-        'I am chunk 3 of doc 3',
+        ' I am chunk 3 of doc 3',
     ]
 
     search_flow = Flow().load_config('flow-query.yml')
     with search_flow:
-        search_flow.search(input_fn=search_data, output_fn=validate_granularity_1, callback_on_body=True, granularity=1)
+        search_flow.search(
```

Comment:
We should remove then `granularity` from the `request` generator function. With the traversal path it makes no sense anymore right?

Reply:
:+1:

Reply:
u do it here or want to merge this and open a new one?